File: src/main/kotlin/java_code/DecisionMaking.java
package java_code;


import java.util.Scanner;

// TODO: Md. Al Amin (01750349979)
public class DecisionMaking {
    public static void main(String[] args) {
        int a = 10;
        int b = 15;

        if (a > b){
            System.out.println("a is greater than b.");
        }else if(a < b){
            System.out.println("a is less than b.");
        }else{
            System.out.println("a is equal than b.");
        }
        String result = (a==b) ? "yes" : "no";
    }


    public static void switchCase(){
        int a = 12;
        int b = 8;
        Scanner scanner = new Scanner(System.in);
        String operator = scanner.nextLine();
        String result = switch (operator) {
            case "+" -> "%d + %d = %d".formatted(a,b,(a+b));
            case "-" -> "%d - %d = %d".formatted(a,b,(a-b));
            case "*" -> "%d * %d = %d".formatted(a,b,(a*b));
            case "/" -> "%d / %d = %d".formatted(a,b,(a/b));
            default -> "%s is invalid".formatted(operator);
        };
        System.out.println(result);
    }

    public void loop(){

        for (int i = 0; i <= 10; i++){
            System.out.println(i);
        }
        for (int i = 0; i <= 10; i = i+2){
            System.out.println(i);
        }
        for (int i = 10; i >= 0; i = i -2){
            System.out.println(i);
        }



        int x = 10;
        while (x-- > 0){
            if (x==6){
                x--;
                //break
                continue;
            }
            System.out.println(x);
            x--;
        }

    }


    public void exception(){


        int m = 10;
        int n = 0;
        try {
            int t = m/n;
        } catch (ArithmeticException e) {
            // handler
        } finally {
            // optional finally block
        }


    }











}

File: src/main/kotlin/variable_declare/DecisionMaking.kt
package variable_declare


// TODO: Md. Al Amin (01750349979)
fun main() {
    var a: Int = 10;
    var b: Int = 15;
    println("Int max value: ${Int.MAX_VALUE}");
    println("Int min value: ${Int.MIN_VALUE}");
    if (a > b){
        println("a is greater than b.");
    }else if(a < b){
        println("a is less than b.");
    }else{
        println("a is equal than b.");
    }

    val result:String = if (a==b) "yes" else "no";
}


class DecisionMaking {

    public fun returnIf() {
        var a: Int = 10;
        var text = if (a in 9..20){
            println("$a is greater than 9.");
            "This number is greater than 9."
        }else{
            println("$a is less than 9.");
            "This number is less than 9."
        }
    }


     fun switchCaseConvertWhen(){
        val a = 12
        val b = 5
        println("Enter operator either +, -, * or /")
        val operator = readLine()
        val result:String = when (operator) {
            "+" -> "$a + $b = ${a + b}";
            "-" -> "$a - $b = ${a - b}";
            "*" -> "$a * $b = ${a * b}";
            "/" -> "$a / $b = ${a / b}";
            else -> "$operator is invalid";
        }
        println(result);
    }
}